### PR TITLE
update IL production capacities

### PR DESCRIPTION
**Pull request by yuvalabou**
Update some of Israel production capacities based on info from [here](https://www.lnrg.technology/2020/08/30/overview-of-the-israeli-electricity-market-2020/) and other sources
"unknown" includes all power plant with no breakdown..

### CODE REVIEW

**Comment by madsnedergaard** [comment]
Would you mind updating [https://github.com/tmrowco/electricitymap-contrib/blob/master/DATA_SOURCES.md](https://github.com/tmrowco/electricitymap-contrib/blob/master/DATA_SOURCES.md) and listen those "other sources" as well?

**Comment by madsnedergaard** [comment]
Two things:

1. Can you provide more direct link to the source of these numbers? I can't seem to find anything directly stating those, so if you calculated it, it would be nice to explain how, since the numbers change quite dramatically :)

2. Data sources should have only one entry per country, with a list of sources - see Netherlands as example:
https://github.com/tmrowco/electricitymap-contrib/blob/0f5fa6ae874c6aed7acc148aeb0a0e2bd401c51c/DATA_SOURCES.md#L92-L94

**Comment by madsnedergaard** [comment]
Thanks for the update. 
Would you mind explaining here how you calculated the numbers used in this PR, since they don't appear directly in the article? 

My quick calculations based on the total production split by these percentages gave different numbers:


As far as I can tell from the article, the number you added in "Unknown" is actually the total production capacity of Israel? :)

**Comment by yuvalabou** [comment]
Yes this is the total production as there is no breakdown for it when we scrape for production or consumption

**Comment by madsnedergaard** [comment]
> Yes this is the total production as there is no breakdown for it when we scrape for production or consumption

The "unknown" capacity should only for the "spare" capacity we can't identify as a specific source - so `unknown = total_capacity - (coal+gas+solar etc.)`

**Comment by madsnedergaard** [comment]
I still don't understand your numbers - would you mind explaining how you calculated the numbers used in this PR?

**Comment by veqtrus** [comment]
The parser takes data from https://www.iec.co.il/en/pages/default.aspx which does not break down production by source. At the moment the parser seems broken as it returns the sum of consumption and reserves (hence always the same value).

The parser should be fixed and IMO 'unknown' capacity should be left undefined.

**Comment by leonkraversky** [comment]
Hello guys, I'm the owner of LNRG Technology. How can i help to integrate the data? Perhaps json format file at LNRG site to help the direct integration?

**Comment by leonkraversky** [comment]
> The parser takes data from https://www.iec.co.il/en/pages/default.aspx which does not break down production by source. At the moment the parser seems broken as it returns the sum of consumption and reserves (hence always the same value).
> 
> The parser should be fixed and IMO 'unknown' capacity should be left undefined.
> 
> Edit: Alternatively the parser should be removed as it can't be used to estimate CO2 intensity.

Note that the IEC controls 78% of the Israeli market (IEC controls all transmission), the rest being self-consumption+off-grid and P2P sales. IEC is hence most of the market (78%), but not all of it.

**Comment by leonkraversky** [comment]
> Thanks for the update.
> Would you mind explaining here how you calculated the numbers used in this PR, since they don't appear directly in the article?
> 
> My quick calculations based on the total production split by these percentages gave different numbers:
> 
> 
> As far as I can tell from the article, the number you added in "Unknown" is actually the total production capacity of Israel? :)

This is not generation, but installed capacity (measured in MWp or GWp). Generation (measured in kWh) is distributed differently.

**Comment by veqtrus** [comment]
@leonkraversky this PR is about updating the installed capacities. The issue is that we currently don't have access to generation statistics by source, only the total generation, hence we report it as "unknown", with its carbon intensity calculated based on yearly IEA generation statistics. We would of course prefer to have real-time generation by source in order to accurately estimate CO2 emissions.

**Comment by leonkraversky** [comment]
> @leonkraversky this PR is about updating the installed capacities. The issue is that we currently don't have access to generation statistics by source, only the total generation, hence we report it as "unknown", with its carbon intensity calculated based on yearly IEA generation statistics. We would of course prefer to have real-time generation by source in order to accurately estimate CO2 emissions.

Generation statistics by source is estimated by LNRG in the above mentioned report. It is not available in the executive summary, but can be accessed upon request for integration in to Electricity Map. This is why I asked for the format to get this with some parser.

**Comment by madsnedergaard** [comment]
> > @leonkraversky this PR is about updating the installed capacities. The issue is that we currently don't have access to generation statistics by source, only the total generation, hence we report it as "unknown", with its carbon intensity calculated based on yearly IEA generation statistics. We would of course prefer to have real-time generation by source in order to accurately estimate CO2 emissions.
> 
> Generation statistics by source is estimated by LNRG in the above mentioned report. It is not available in the executive summary, but can be accessed upon request for integration into Electricity Map open source project (I would approve this). This is why I asked for the format of data, to get this from LNRG website with some parser.

Hi @leonkraversky, thanks for joining the conversation here and offering your help! 💪 

The best possible thing, as @veqtrus said, is if it was possible for us (perhaps with a secret API token) to get a "real-time" breakdown of production sources - you can see more about our [requirements for data here](https://github.com/tmrowco/electricitymap-contrib/wiki/Requirements-for-data). It would of course also be nice with just a breakdown of source capacities, but do bear in mind that this data will be part of the open-source repository here and can be accessed by anyone :)

**Comment by veqtrus** [comment]
JSON would likely be our preferred format. A lot of sources offer access to historical data, however this is not required. It would be preferable if the source would return an array of data points and not just the latest one in case we miss some, but again this is not required. The format could be like:

```json
[
  {
    "datetime": "2021-06-23 14:30",
    "production": {
      "coal": 1010,
      "gas": 2020
    }
  },
  {
    "datetime": "2021-06-23 15:00",
    "production": {
      "coal": 1030,
      "gas": 2050
    }
  }
]
```

You can take a look at https://github.com/tmrowco/electricitymap-contrib/wiki/Building-a-new-parser for how we format the data internally.

The Tomorrow team will have to contact you to arrange any API keys if you don't want this endpoint to be publicly accessible.

**Comment by leonkraversky** [comment]
The only thing i can offer to do quick and dirty and uploading a JSON file with annual resolution data for open access on LNRG website (no secured API for this specific matter at this point, but the data is not a "big secret"). Perhaps there would be a more detailed data in the future (monthly or even daily), but i guess that would have to be a more restricted professional access with API tocken etc. 

Will annual data in JSON format as defined above by @veqtrus help this project? (it will be updated once a year for now)

**Comment by skovhus** [comment]
@leonkraversky thanks for offering your help here, we really appreciate that.

But do note that we [require the data](https://github.com/tmrowco/electricitymap-contrib/wiki/Requirements-for-data) to be updated in real-time with less than 2 hours of delay + have a granularity of hourly or higher.

**Comment by madsnedergaard** [comment]
> @leonkraversky thanks for offering your help here, we really appreciate that.
> 
> But do note that we [require the data](https://github.com/tmrowco/electricitymap-contrib/wiki/Requirements-for-parser-data) to be updated in real-time with less than 2 hours of delay + have a granularity of hourly or higher.

With that said it would still be nice to get the latest numbers for **production capacities by source** (so not real-time data of actual production) so we can show the correct numbers here:
Screenshot 2021-06-24 at 09 47 18

If you could add a JSON file with those numbers somewhere on your website (so we can link to it) that would be great :)

**Comment by leonkraversky** [comment]
> > @leonkraversky thanks for offering your help here, we really appreciate that.
> > But do note that we [require the data](https://github.com/tmrowco/electricitymap-contrib/wiki/Requirements-for-parser-data) to be updated in real-time with less than 2 hours of delay + have a granularity of hourly or higher.
> 
> With that said it would still be nice to get the latest numbers for **production capacities by source** (so not real-time data of actual production) so we can show the correct numbers here:
> Screenshot 2021-06-24 at 09 47 18
> 
> If you could add a JSON file with those numbers somewhere on your website (so we can link to it) that would be great :)

Capacity data can be done for the project - it is a public info. Will update you when the JSON is uploaded.

**Comment by leonkraversky** [comment]
> > > @leonkraversky thanks for offering your help here, we really appreciate that.
> > > But do note that we [require the data](https://github.com/tmrowco/electricitymap-contrib/wiki/Requirements-for-parser-data) to be updated in real-time with less than 2 hours of delay + have a granularity of hourly or higher.
> > 
> > 
> > With that said it would still be nice to get the latest numbers for **production capacities by source** (so not real-time data of actual production) so we can show the correct numbers here:
> > Screenshot 2021-06-24 at 09 47 18
> > If you could add a JSON file with those numbers somewhere on your website (so we can link to it) that would be great :)
> 
> Capacity data can be done for the project - it is a public info. Will update you when the JSON is uploaded.

Done uploading capacities here: https://www.lnrg.technology/app/download/12390579357/IL+electricity+capacities.json?t=1624618958

The file capacity data is for the date of 01-01-2020.

Please check whether the format is fine, not sure about zone_key='IL'.

**Comment by madsnedergaard** [comment]
Great, thanks for the file! @leonkraversky 
@yuvalabou Would you mind using these new numbers for the capacities in zones.json and adding the new link to `DATA_SOURCES.md`? :)

```
    "capacity": {
      "coal": 4840,
      "gas": 12776,
      "solar": 1962,
      "wind": 28,
      "hydro": 6.72,
      "hydro storage": 150.0,
      "biomass" : 27,
      "nuclear": 0,
      "oil": 44,
      "geothermal": 0,
      "unknown": 175
    }
```

**Comment by leonkraversky** [comment]
> Great, thanks for the file! @leonkraversky
> @yuvalabou Would you mind using these new numbers for the capacities in zones.json and adding the new link to `DATA_SOURCES.md`? :)
> 
> ```
>     "capacity": {
>       "coal": 4840,
>       "gas": 12776,
>       "solar": 1962,
>       "wind": 28,
>       "hydro": 6.72,
>       "hydro storage": 150.0,
>       "biomass" : 27,
>       "nuclear": 0,
>       "oil": 44,
>       "geothermal": 0,
>       "unknown": 175
>     }
> ```

You can source the data directly from that file if you like. It shall be updated once a year, but that is better than nothing.
Regarding capacity classification, unknown refers to diesel and kerogen - I was not sure whether those can be put under the "oil" section. What do you think?

File: DATA_SOURCES.md [hunk 1]
```diff
@@ -54,7 +54,9 @@ Real-time electricity data is obtained using [parsers](https://github.com/tmrowc
 - Hungary: [ENTSOE](https://transparency.entsoe.eu/content/static_content/Static%20content/web%20api/Guide.html)
 - Iceland: [LANDSNET](https://amper.landsnet.is/generation/api/Values)
 - Ireland: [ENTSOE](https://transparency.entsoe.eu/content/static_content/Static%20content/web%20api/Guide.html)
-- Israel: [IEC](https://www.iec.co.il/en/pages/default.aspx)
+- Israel:
+  - [IEC](https://www.iec.co.il/en/pages/default.aspx)
+  - [LNRG](https://www.lnrg.technology/2020/08/30/overview-of-the-israeli-electricity-market-2020/)
```
Comment (madsnedergaard) [inline DATA_SOURCES.md:59]: This change to "Real-time electricity data sources" should be reverted :)

**Review by madsnedergaard — COMMENTED** [review]
This should be the *very last* comment! 😄

**Comment by madsnedergaard** [comment]
> Regarding capacity classification, unknown refers to diesel and kerogen - I was not sure whether those can be put under the "oil" section. What do you think?

I think this is fine :)

File: DATA_SOURCES.md (hunk 1 above)
Comment (madsnedergaard) [inline DATA_SOURCES.md:59]: Making the change here so we can get this PR merged! :)
```suggestion
- Israel: [IEC](https://www.iec.co.il/en/pages/default.aspx)
```

**Review by madsnedergaard — APPROVED** [review]

**Pull request merged**